app/models/event.rb

```diff
@@ -22,7 +22,7 @@ class Event < ApplicationRecord
   default_scope       -> { where(label: Whitelabel[:label_id]) }
 
   scope :with_topics, -> { joins(:topics).distinct }
-  scope :current,     -> { where(date: Date.today.to_time..(Time.now + 9.weeks)).limit(1).order('date ASC') }
+  scope :current,     -> { where(date: Date.today.to_time..).limit(1).order('date ASC') }
```

Comment:
is this an unlimited range? if it is, this is suboptimal, use a `>` operator instead

Reply:
Yes it's an unlimited range. Activerecord handles that for us:

```ruby
Event.current.to_sql
=> "SELECT \"events\".* FROM \"events\" WHERE \"events\".\"label\" = 'hamburg' AND \"events\".\"date\" >= '2024-12-02 23:00:00' ORDER BY date ASC LIMIT 1"
```

Why do you feel like this is suboptimal? We can change this of course, I'm  just wondering.

Reply:
That's pretty nice of AR to optimize it for us. Anyways, given the rule of least surprise I would argue that its better and has less cognitive overhead to use `>` instead.

Reply:
>= as it seems :-D

Reply:
Mh I get your point. I get it's something you need to get used to. Personally I always struggle with date comparisons, so I actually find the unlimited range more intuitive. But whatever, I changed the code to use `>=` instead. This also matches the `latest` scope so I guess it better matches the current code style.

Reply:
In my mind, a range translates to a `BETWEEN` query. So it's kind of akward to use it here. I would also complain if you would use an unlimited range to compare dates in ruby.

Reply:
I guess we just have different mental models on this which is fine!

What I meant by comparisons is that I often find myself wondering for a second what exactly it means for a date to be greater than another date. So if I see a query like this:

```ruby
Event.where('date > ?', Time.current)
```
I have to think a second on whether the found events will be in the future or in the past. But that's probably a me-problem and I don't really have strong feelings about this. Just my two cents on the topic and why I chose to implement it that way in the first place :sweat_smile:

Reply:
just out of curiosity, you would rather write `(3.years.ago..).cover?(Date.today)` than `3.years.ago <= Date.today`

Reply:
oh wow, had a look into the code first, and I would expect the usage of an open range instead of hand made SQL, as you get better results when combining scopes with joins.

Reply:
Not sure about this. If we are in Rails I'd probably prefer `3.years.ago.before?(Date.today)` or just `3.years.ago.past?` if we want to check against `Time.current` and not `Date.today`.

Otherwise I'm not sure. Probably by habit I'd use the normal comparison.

It's a bit different to the AR query though. The order  is different. The equivalent ruby code would be more like this:
```ruby
date > Time.current # or
date.in?(Time.current..)
``` 
And for this I'd probably prefer the latter variant but it's not something I'd argue about and I've probably written the first variant a lot of times as well.

In the context of AR though it feels like we work around the abstractions of AR by falling back to using "handwritten" SQL. This also has some disadvantages like missing quoting on the column name for example. Maybe this is the real reason why I'd choose `where(date: Time.current..)` over `where('date > ?', Time.current)` at any time. It just feels better to do it the way AR intends it to be, I don't know :sweat_smile:

Reply:
the notorious ambigous name problem is a good argument for using the range. it's also documented as good practice in the guides https://guides.rubyonrails.org/active_record_querying.html#range-conditions 

i find the syntax and the choice to support it questionable, but i guess it was an easy fix instead of properly parsing the provided "handwritten sql" string or having to write `Event.where(Event.arel_table[:date].gt(Time.current))`

Reply:
I‘m guilty of using internal API (arel_table) in the past to prevent ambiguous names. 

But overall the approach to use open ranges feels quite rubyish. Still takes a while to get used to.

Reply:
Feels like we settled on the range variant, so I dropped the commit that converted it to the handwritten SQL. Let me know if there is something else we need to do to get this merged :)